### ISSUE TITLE
Automatic furnace refueling only refuels one furnace

### ISSUE THREAD

**Issue by darienac** [SquidDev/artist#7]
When coal is placed in the turtle, it picks a random furnace to get all the coal instead of distributing it throughout all the furnaces evenly.

There should also be a priority to place coal within furnaces that have items ready to be smelted and an appropriate amount of items should be put in for that task (minding the -1 item smelting bug, so 16 items would actually need 3 coal, but 15 items would need 2)

**Comment by SquidDev** [comment]
Yeah, this is probably a good idea. I've not really thought about this before as the refueller rather assumes an abundance of coal. 

As far as only refuelling as much as needed, I'm less sure. One could definitely do some neat stuff with optimal fuel choosing, etc... but it's probably overcomplicated for where Artist is right now. I'd like to get the other basics in first.
